app/payment/session.go

```diff
@@ -90,7 +90,7 @@ func RespondPayChProposal(pctx context.Context, s perun.SessionAPI, proposalID s
 }
 
 // CloseSession closes the current session.
-func CloseSession(s perun.SessionAPI, force bool) ([]PayChInfo, error) {
+func CloseSession(s perun.SessionAPI, force bool) ([]PayChInfo, perun.APIErrorV2) {
```

Comment:
You should also do type assertion on the error message. If it is "Failed Precondition Error" with open channels, then the channels info should be converted to payment channels info.